### PR TITLE
merge documentation branch into master

### PR DESCRIPTION
**Pull request by reece**
This is the moment we've all been waiting for!

Major changes:
- new documentation outline
- significant new content and editing
- semi-automated doc builds (make docs)

**Terrific community effort everyone!**

The documentation branch will be retired after this merge.
(I think this is done with git push upstream --delete documentation... but let's wait until the merge!)

I have verified that docs build correctly from this PR.

### CODE REVIEW

**Comment by calbach** [comment]
Thanks Reece.

A few high level notes:
- It looks a large number of the changes are in the auto-generated docs, so possibly there were some changes made to the doc generation code? It would reduce the noise to send a PR which updates the autogenerated docs ahead of this (though it's not strictly necessary, if everyone understands that this directory is autogenerated).
- We should squash probably the commits down.

File: doc/source/schemas/intro.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Introduction
```
Comment (calbach) [inline doc/source/schemas/intro.rst]: This file does not seem useful. Is this autogenerated?

File: doc/source/appendix/samplecode_intro.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+.. _samplecode:
+
+*******************
+Code samples (stub)
```
Comment (calbach) [inline doc/source/appendix/samplecode_intro.rst]: I'm suspicious of this file since it seems pretty generic and has 'stub' in the title. Does any of this pertain to the GA4GH APIs?

File: doc/Makefile-broken [hunk 3]
```diff
@@ -0,0 +1,178 @@
+# Makefile for Sphinx documentation
```
Comment (dglazer) [inline doc/Makefile-broken]: This filename is confusing -- if it's truly broken we should delete it; if it's needed for mvn, as the comments imply, I'd suggest renaming it to something like Makefile-for-mvn.

File: doc/README.rst [hunk 4]
```diff
@@ -5,9 +5,16 @@ This directory contains source and tools to build the GA4GH schemas
...
 The documentation is built automatically by Read the Docs upon commit
-to the master branch.  The resulting docs should appear at
-https://ga4gh-schemas.readthedocs.org/ within 2 minutes after commit.
+to the master and documentation branches.  See:
 
+The `resulting docs <http://ga4gh-schemas.readthedocs.org/>`_ are
+built automatically at Read the Docs, usually within two minutes of a
+commit.
+
+.. note:: While documentation is under development in the
```
Comment (dglazer) [inline doc/README.rst]: This note can go away, since there won't be a documentation branch after we merge.

**Comment by dglazer** [comment]
doc/source/_static/ga4gh-objects.png covers objects that aren't yet part of master -- we should either remove it (if it's not referenced) or update it (if it is).

**Comment by dglazer** [comment]
Thank you Reece -- this mostly looks great. I added a couple of small comments that can be addressed either before or after merge.

I agree with @calbach that the changes to the `doc/source/schemas/*.rst` auto-generated files are hard to parse. As long as we're clear that those are all auto-generated, and the only changes that need review are to the .avdl files themselves, I'm okay for now. (And eager to see us get rid of the need to check them in at all.)

**Comment by diekhans** [comment]
Go ahead and remove _static/ga4gh-objects.png ; it's a
conceptual drawing I create.  It isn't referenced and the source
isn't checked in.

We can add it in later if needed.

File: doc/source/schemas/intro.rst (hunk 1 above)
Comment (reece) [inline doc/source/schemas/intro.rst]: Fixed; commit pending.

File: doc/Makefile-broken (hunk 3 above)
Comment (reece) [inline doc/Makefile-broken]: Fixed; commit pending.

**Comment by reece** [comment]
Removed doc/source/_static/ga4gh-objects.png

File: doc/README.rst (hunk 4 above)
Comment (reece) [inline doc/README.rst]: Agreed. Commit pending.

**Comment by reece** [comment]
This merge to master was premature. Closing it unmerged.

The above suggestions are reflected in PR #503 against the *documentation* branch. Once that branch is in shape, it will be merged into master.
